packages/providers/email-mailgun/lib/index.js

```diff
@@ -1,21 +1,42 @@
 'use strict';
 
-const mailgunFactory = require('mailgun-js');
+const formData = require('form-data');
+const Mailgun = require('mailgun.js');
 const { removeUndefined } = require('@strapi/utils');
 
+const optionsMap = {
+  apiKey: { field: 'key', fn: value => value },
+  host: { field: 'url', fn: value => `https://${value || 'api.mailgun.net'}` },
+};
+
 module.exports = {
+  convertProviderOptions(providerOptions = {}) {
+    const newOptions = {};
+    if (typeof providerOptions === 'object') {
+      Object.keys(providerOptions).forEach(key => {
+        if (Object.keys(optionsMap).includes(key)) {
+          newOptions[optionsMap[key].field] = optionsMap[key].fn(providerOptions[key]);
+        } else {
+          newOptions[key] = providerOptions[key];
+        }
+      });
+    }
+    return newOptions;
+  },
+
   init(providerOptions = {}, settings = {}) {
-    const mailgun = mailgunFactory({
-      mute: false,
-      ...providerOptions,
-    });
+    const defaults = {
+      username: 'api',
+    };
+    const mailgun = new Mailgun(formData);
+    const mg = mailgun.client({ ...defaults, ...this.convertProviderOptions(providerOptions) });
 
     return {
       send(options) {
         return new Promise((resolve, reject) => {
```

Comment:
sends mg.messages returns a promise you can get ride of this one entirely it seems ?

Reply:
@alexandrebodin thank you for the review. I am addressing this right now. Should have something ready in an hour or so (need to test that everything works in an example strapi deployment). And, yes, please push directly to this branch if needed.

Reply:
@alexandrebodin adjusted the code as per your suggestion. Good catch, thank you.